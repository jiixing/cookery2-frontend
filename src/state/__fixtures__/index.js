import recipes from './recipes'
import * as rest from './rest'

const recipe = () => rest.recipe
const state = { recipes: recipes(), recipe: recipe() }

export { recipes, recipe, rest }
export default () => state
