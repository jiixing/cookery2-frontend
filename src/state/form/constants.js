export const namespace = 'form'
export const recipeModel = 'recipe'

export const RECIPE_CREATE_SUCCESS = 'recipes:recipe:create:success'
export const RECIPE_CREATE_FAIL = 'recipes:recipe:create:fail'
