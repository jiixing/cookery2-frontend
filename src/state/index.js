import { combineReducers } from 'redux'

import {
  constants as recipesConstants,
  reducer as recipesReducer,
} from './recipes'
import {
  constants as recipeConstants,
  reducer as recipeReducer,
} from './recipe'

const rootReducer = combineReducers({
  [recipesConstants.namespace]: recipesReducer,
  [recipeConstants.namespace]: recipeReducer,
})

export default rootReducer
