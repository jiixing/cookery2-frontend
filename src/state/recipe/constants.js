export const namespace = 'recipe'

export const REQUEST_RECIPE = `${namespace}:request`
export const RECEIVE_RECIPE = `${namespace}:receive`
