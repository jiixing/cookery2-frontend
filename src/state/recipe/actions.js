// @flow

import { fetch } from 'rest'

import { REQUEST_RECIPE, RECEIVE_RECIPE } from './constants'

import type { RecipeDetail } from 'types'

const requestRecipe = () => ({
  type: REQUEST_RECIPE
})

const receiveRecipe = (recipe: RecipeDetail) => ({
  type: RECEIVE_RECIPE,
  payload: recipe
})

export function fetchRecipe (id: number) {
  return (dispatch: Dispatch) => {
    dispatch(requestRecipe())

    return fetch(`/recipes/${id}`)
      .then(response => response.json())
      .then(recipe => dispatch(receiveRecipe(recipe)))
  }
}
