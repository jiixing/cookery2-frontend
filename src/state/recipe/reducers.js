import { REQUEST_RECIPE, RECEIVE_RECIPE } from './constants'

const initialState = { recipe: {} }

function recipe(state = initialState, action) {
  switch (action.type) {
    case REQUEST_RECIPE:
      return initialState
    case RECEIVE_RECIPE:
      return action.payload
    default:
      return state
  }
}

export default recipe
