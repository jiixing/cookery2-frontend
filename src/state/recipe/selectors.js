import * as R from 'ramda'

import { namespace } from './constants'

const recipeSelector = R.path([namespace])
export default recipeSelector
