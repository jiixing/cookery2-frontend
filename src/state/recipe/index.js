import * as constants from './constants'
import * as actions from './actions'

export { actions, constants }
export { default as recipeSelector } from './selectors'
export { default as reducer } from './reducers'
