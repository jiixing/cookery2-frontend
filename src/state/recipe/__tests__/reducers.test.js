import { recipe as recipeState } from 'state/__fixtures__'

import { REQUEST_RECIPE, RECEIVE_RECIPE } from '../constants'
import recipe from '../reducers'

describe('recipe reducer', () => {
  it('returns the initial state', () => {
    expect(recipe(undefined, {})).toMatchSnapshot()
  })

  it('handles REQUEST_RECIPE', () => {
    const action = {
      type: REQUEST_RECIPE
    }

    expect(recipe(recipeState(), action)).toMatchSnapshot()
  })

  it('handles RECEIVE_RECIPE', () => {
    const action = {
      type: RECEIVE_RECIPE,
      payload: recipeState()
    }

    expect(recipe(undefined, action)).toMatchSnapshot()
  })
})
