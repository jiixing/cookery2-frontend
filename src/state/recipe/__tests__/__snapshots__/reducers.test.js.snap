// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`recipe reducer handles RECEIVE_RECIPE 1`] = `
Object {
  "difficulty": "easy",
  "duration": 0,
  "id": 1,
  "name": "test name",
  "quantities": Array [
    Object {
      "amount": 50,
      "name": "sugar",
      "unit": "gr",
    },
    Object {
      "amount": 1000,
      "name": "water",
      "unit": "ml",
    },
    Object {
      "amount": 2,
      "name": "salt",
      "unit": "pinch",
    },
  ],
  "servings": 2,
  "steps": Array [
    Object {
      "description": "Mix all the ingredients",
      "isRecipe": false,
    },
    Object {
      "id": 69,
      "isRecipe": true,
      "name": "Other",
    },
  ],
}
`;

exports[`recipe reducer handles REQUEST_RECIPE 1`] = `
Object {
  "recipe": Object {},
}
`;

exports[`recipe reducer returns the initial state 1`] = `
Object {
  "recipe": Object {},
}
`;
